drivers/flash/flash_sam0.c

```diff
@@ -417,11 +434,7 @@ static const struct flash_driver_api flash_sam0_api = {
 #ifdef CONFIG_FLASH_PAGE_LAYOUT
 	.page_layout = flash_sam0_page_layout,
 #endif
-#if CONFIG_SOC_FLASH_SAM0_EMULATE_BYTE_PAGES
-	.write_block_size = 1,
-#else
-	.write_block_size = FLASH_PAGE_SIZE,
-#endif
+	.get_parameters = flash_sam0_get_parameters,
```

Comment:
I have a slight preference for having the unconditional fields appear before any conditional fields, so would put this before `.page_layout`.  Just a suggestion.

Reply:
My personal preference is to initialize fields, if possible, in the order of appearance. This makes easier for me to figure out if I have missed something, etc. Looks messier when a lot of c preprocessor is involved but still easier to figure out. Do you want me to change it?

Reply:
Initializing in the order of definition isn't just a preference; it's a requirement for C++.  What I meant was to move the `get_parameters` definition up next to the other ones, and keep the optional ones at the end.  Since it's new, it doesn't matter whether it goes at the end or is inserted: the relative order of everything else remains unchanged.

Reply:
> Do you want me to change it?

Yes, but I'm not going to block if you don't.

Reply:
OK I will do this.

Reply:
For C99 the order of definition of structure fields, when using "designators", does not matter.